### DISCUSSION TITLE
Access map from different bpftrace scripts

### DISCUSSION THREAD

**Discussion by nela** [bpftrace/bpftrace#2495] [Q&A] [answered]
Is is possible to access the same map from different bpftrace scripts? Using the bpftool I can se that the map is created and in memory, but unable to make two different bpftrace scripts read and update the map.

**Comment by viktormalik** [comment, marked as answer]
This is not possible at the moment. Bpftrace scripts always create their own maps and cannot reuse maps existing in the system.

**Reply by nela** [reply]
Alright, thanks for the quick reply.

**Reply by fbs** [reply]
Can you tell more about your use case for this. For a good use case we can atleast open an issue to discuss what it would look like.

**Reply by nela** [reply]
Working with userland objects via uprobes (often) requires includes of system wide headers, but these will conflict with kernel types for kprobes. The solution is therefore to keep them in their own scripts. However, that leaves no possibility for the probes in different scripts to communicate via maps.

This is a part of my masters thesis, where I am looking into using bpf for [runtime verification](https://en.wikipedia.org/wiki/Runtime_verification). The case is to trace an nginx webserver with corresponding tcp syscalls, and implement a state machine by use of probes and maps, for identifying any anomalies.

I am not entirely certain whether bpf was intented for a use such as this. But I do imagine similar usecase for map access across scripts coming up.
